lib/private/Comments/Comment.php

```diff
@@ -220,7 +220,7 @@ public function setMessage($message, $maxLength = self::MAX_MESSAGE_LENGTH): ICo
 	 *
 	 */
 	public function getMentions(): array {
-		$ok = preg_match_all("/\B(?<![^a-z0-9_\-@\.\'\s])@(\"guest\/[a-f0-9]+\"|\"group\/[a-z0-9_\-@\.\' ]+\"|\"[a-z0-9_\-@\.\' ]+\"|[a-z0-9_\-@\.\']+)/i", $this->getMessage(), $mentions);
+		$ok = preg_match_all("/\B(?<![^a-z0-9_\-@\.\'\s])@(\"guest\/[a-f0-9]+\"|\"(?:federated_)?(?:group|team|user){1}\/[a-z0-9_\-@\.\' \/:]+\"|\"[a-z0-9_\-@\.\' ]+\"|[a-z0-9_\-@\.\']+)/i", $this->getMessage(), $mentions);
```

Comment:
Nit: 
I think you can remove some `\` since `.` lose it's special meaning between `[ ]`
```suggestion
		$ok = preg_match_all("/\B(?<![^a-z0-9_\-@.'\s])@(\"guest\/[a-f0-9]+\"|\"(?:federated_)?(?:group|team|user){1}\/[a-z0-9_\-@.' \/:]+\"|\"[a-z0-9_\-@.' ]+\"|[a-z0-9_\-@.']+)/i", $this->getMessage(), $mentions);
```